### PR TITLE
Remove Security alerts.

### PR DESCRIPTION
**Pull request by ThatOneNovaGuy**
Created SECURITY.md file.

Adjusted workflow/*.yml files to OSSF recommendations.

### CODE REVIEW

**Review by ThatOneNovaGuy — COMMENTED** [review]
SECURITY.md will be refined at a later date, pushing for now.

**Review by JeffreyBlinksma — DISMISSED** [review]

**Review by ghost — COMMENTED** [review]
Looks good

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -38,11 +38,11 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@d0651293c4a5a52e711f25b41b05b2212f385d28
```
Comment (PeskyPotato) [inline .github/workflows/codeql-analysis.yml:44]: Is this the commit hash?

It gives a 404.
https://github.com/actions/checkout/commit/d0651293c4a5a52e711f25b41b05b2212f385d28

v3 commit hash looks to be `2541b1294d2704b0964813337f33b291d3f8596b`. Let me know if I'm reading this right. :slightly_smiling_face:

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -38,11 +38,11 @@ jobs:
...
 
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
-      uses: github/codeql-action/init@v2
+      uses: github/codeql-action/init@5a54a995785650c9dbc4c88cc1e1d1b4df4302a8
```
Comment (PeskyPotato) [inline .github/workflows/codeql-analysis.yml:48]: All of the codeql-action commit hashes give out a 404. The commit hash for v2 seems to be 3e7e3b32d0fb8283594bb0a76cc60a00918b0969.

https://github.com/github/codeql-action/commit/5a54a995785650c9dbc4c88cc1e1d1b4df4302a8

**Review by PeskyPotato — CHANGES_REQUESTED** [review]
Just wanted to clarify the commit hashes for the release tags are correct. I can't locate them in the orignal repo's, let me know if I'm looking at it wrong. Thanks! : )

**Comment by ThatOneNovaGuy** [comment]
Stale branch, should fix most issues by resolving other PR's.

**Pull request closed without merging**